### ISSUE TITLE
[FEATURE] Improved event filters based on dates

### ISSUE THREAD

**Issue by sandeep-ps** [rokwire/illinois-app#83]
**Is your feature request related to a problem? Please describe.**
Currently, when using event filters like today, next 7 days, etc. from the app, because of the current query parameters being used, we display only those events that start and end during a specific period. We are not displaying events that started in the past and ends in the specified period (today, next 7 days, etc.) or continues into the future beyond the specified period. Those events are live during that period and needs to be displayed.

**Describe the solution you'd like**
We have added new query parameters to Events Building Block. Currently it's being deployed to api-dev and on Monday we will request deployment to TEST and PROD.

`startDate.lte`: Start date of the event is less than or equal to give input date
`startDate.gte`: Start date of the event is greater than or equal to give input date
`endDate.lte`: End date of the event is less than or equal to give input date
`endDate.gte`: Start date of the event is greater than or equal to give input date

Using these new parameters, we can do better events search based on dates:

If a user selects the following options at 3 pm central (8 pm GMT) on Sep 18, using the specific queries in each option, we can get all the live events during that specific period: 

1. Upcoming:
`?endDate.gte=2020-09-18T:20:00:00`

2. Today (09/18):
`?startDate.lte=2020-09-19T:04:59:59&endDate.gte=2020-09-18T:20:00:00` 
End time is 04:59:59 on next day because of conversion to GMT. I think the current implementation should also be using this time.

3. Next seven days (09/18 - 09/24):
`?startDate.lte=2020-09-25T:04:59:59&endDate.gte=2020-09-18T:20:00:00`

4. This weekend (09/19 - 09/20):
`?startDate.lte=2020-09-21T:04:59:59&endDate.gte=2020-09-19T:05:00:00`

5. Next 30 days (09/18-10/18):
`?startDate.lte=2020-10-18T:23:59:59&endDate.gte=2020-09-18T:15:00:00`

**Describe alternatives you've considered**
No other alternatives were considered at this time.

**Additional context**
This is related to a bug that super events do not get displayed in the event listing after they start. #49

**Comment by mdryankov** [comment]
@sandeep-ps I'm unable to test the desired changes. Dev looks empty. Prod looks not deployed yet. Could you please take a look?

**Comment by sandeep-ps** [comment]
@mdryankov, we have fixed the issue in api-dev. This query should get you some example data: 
```
curl --location --request GET 'https://api-dev.rokwire.illinois.edu/events?category=Academic&startDate.gte=2020-01-01T00:00:00&startDate.lte=2020-06-01T00:00:00' \
--header 'ROKWIRE-API-KEY: <API Key>'
```

**Comment by sandeep-ps** [comment]
@mdryankov, FYI, these search filters are now also available in the TEST and PROD instances. Thanks.

**Comment by sandeep-ps** [comment]
@mdryankov, implementing the new event filters mentioned here will fix issues like the ones mentioned in #131. Could you please take a look at this feature and let me know if you have any questions? Thanks.

CC: @pmarkhennessy, @kpecknold

**Comment by mdryankov** [comment]
@sandeep-ps I'm testting with such request and it gives me strange results - please see the screenshot
![device-2020-11-03-150233](https://user-images.githubusercontent.com/7037018/97988485-ba956000-1de5-11eb-868d-8139011a2568.png)

**Comment by sandeep-ps** [comment]
Hi @mdryankov, I see couple of commits linked to this issue. You might have figured this out, but if not, the second event above is running till Aug 25, 2021 as per the event JSON data. Are there still questions about the new search filters? Please let me know. Thanks.

**Comment by mdryankov** [comment]
Thanks @sandeep-ps .  Relaized it a bit later. This functionality has been merged and get part of 2.3.16+416 version. Assigning to you for validation

**Comment by sandeep-ps** [comment]
The current implementation looks good. Thanks.
